### PR TITLE
[crypto] PSA crypto integration

### PR DESCRIPTION
**Pull request by hemanth-silabs**
This PR aligns the radio abstraction with PSA inclusion in the PR openthread/openthread#6862.

Depends-On: openthread/openthread#6862

### CODE REVIEW

File: src/src/openthread-core-efr32-config.h [hunk 1]
```diff
@@ -275,16 +275,6 @@
 #define OPENTHREAD_CONFIG_EFR32_UART_TX_FLUSH_TIMEOUT_MS 500
 #endif
 
-/**
```
Comment (suveshpratapa) [inline src/src/openthread-core-efr32-config.h:278]: @hemanth-silabs I see you've removed this. Who is supposed to set OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE?

File: src/src/openthread-core-efr32-config.h [hunk 2]
```diff
@@ -295,4 +285,8 @@
  */
 #define OPENTHREAD_CONFIG_PSA_ITS_NVM_OFFSET 0x20000
 
+#ifndef OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
```
Comment (suveshpratapa) [inline src/src/openthread-core-efr32-config.h:288]: Who sets this? Can we add a header / comment to this define to describe it?

**Review by suveshpratapa — CHANGES_REQUESTED** [review]
Clarifications requested.

File: src/src/openthread-core-efr32-config.h (hunk 1 above)
Comment (hemanth-silabs) [reply src/src/openthread-core-efr32-config.h:278]: @suveshpratapa we don't rely on `OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE` anymore. We do need another macro, but that is to config the stack itself. Radio platform abstraction need not have it.

File: src/src/openthread-core-efr32-config.h (hunk 2 above)
Comment (hemanth-silabs) [reply src/src/openthread-core-efr32-config.h:288]: This will default to 1 in the config file, once tested and PSA is enabled.

Comment (suveshpratapa) [reply src/src/openthread-core-efr32-config.h:288]: Okay, maybe we can add a comment above it to explain what this option is.

File: src/src/radio.c [hunk 3]
```diff
@@ -364,7 +356,12 @@ static otError radioProcessTransmitSecurity(otRadioFrame *aFrame)
         otMacFrameSetFrameCounter(aFrame, sMacFrameCounter++);
     }
 
+#if defined(_SILICON_LABS_32B_SERIES_2) && OPENTHREAD_RADIO
+    efr32PlatProcessTransmitAesCcm(aFrame, &sExtAddress);
```
Comment (romacdon) [inline src/src/radio.c:360]: Where is efr32PlatProcessTransmitAesCcm defined in GitHub?

File: src/src/openthread-core-efr32-config.h (hunk 1 above)
Comment (romacdon) [reply src/src/openthread-core-efr32-config.h:278]: And just to make sure I'm clear these "same" changes are going to be made to the GSDK version of the PAL, correct?

**Review by romacdon — CHANGES_REQUESTED** [review]

File: src/src/radio.c (hunk 3 above)
Comment (hemanth-silabs) [reply src/src/radio.c:360]: I have added `ieee802154-packet-utils.cpp` to `to-efr32` to enable this support.

File: src/src/openthread-core-efr32-config.h (hunk 1 above)
Comment (hemanth-silabs) [reply src/src/openthread-core-efr32-config.h:278]: @romacdon Yes. These changes will go to GSDK PAL as well. It is already present in a separate branch.

File: src/src/radio.c (hunk 3 above)
Comment (hemanth-silabs) [reply src/src/radio.c]: @romacdon I have removed reference to `efr32PlatProcessTransmitAesCcm` and related cpp file.

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by lmnotran — APPROVED** [review]
LGTM. Thanks for improving `efr32-definitions`!

**Review by romacdon — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**